driver-sync/src/test/functional/com/mongodb/client/AbstractCrudTest.java

```diff
@@ -122,17 +122,9 @@ public void shouldPassAllOutcomes() {
         assumeFalse(definition.getString("description").getValue().startsWith("Deprecated count"));
 
         BsonDocument expectedOutcome = definition.getDocument("outcome", null);
-        // check if v1 test
-        if (definition.containsKey("operation")) {
-            runOperation(expectedOutcome, definition.getDocument("operation"),
-                    expectedOutcome != null && expectedOutcome.containsKey("result") && expectedOutcome.isDocument("result")
-                            ? expectedOutcome.get("result") : null);
-        } else {  // v2 test
-            BsonArray operations = definition.getArray("operations");
-            for (BsonValue operation : operations) {
-                runOperation(expectedOutcome, operation.asDocument(), operation.asDocument().get("result", null));
-            }
-        }
+        runOperation(expectedOutcome, definition.getDocument("operation"),
```

Comment:
I took a closer look at this whole class and end up doing the following:

* inline runOperation method entirely
* default expectedOutcome to BsonDocument instead of null, removing the need for some null checks
* removed ternary expression entirely for initialization of expectedResult, as it doesn't appear to be necessary at all and actually prevents comparisons against non-document expected results, of which there are many.

Reply:
Done

Reply:
👏 thanks!